okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -175,24 +177,31 @@ public void onFailure(Call call, IOException exception) {
                 // Fail call if backoffs are exhausted or if no retry URL can be determined.
                 Optional<Duration> backoff = backoffStrategy.nextBackoff();
                 if (!backoff.isPresent()) {
-                    callback.onFailure(call, new IOException("Failed to complete the request due to an "
-                            + "IOException", exception));
+                    callback.onFailure(call, new SafeIoException(
+                            "Failed to complete the request due to an IOException",
+                            exception,
+                            UnsafeArg.of("requestUrl", call.request().url().toString())));
                     return;
                 }
+
                 Optional<HttpUrl> redirectTo = urls.redirectToNext(request().url());
                 if (!redirectTo.isPresent()) {
-                    callback.onFailure(call, new IOException("Failed to determine valid failover URL for '"
-                            + request().url() + "' and base URLs " + urls.getBaseUrls()));
+                    callback.onFailure(call, new SafeIoException(
+                            "Failed to determine valid failover URL",
+                            exception,
+                            UnsafeArg.of("requestUrl", call.request().url().toString()),
+                            UnsafeArg.of("baseUrls", urls.getBaseUrls())));
                     return;
                 }
 
+                log.info("Retrying call after failure",
```

Comment:
@pkoenig10 why do we think all services want to see this log line all the timme? If you're interested in something specific at one service, why not dial up debug logging for that service?

Reply:
When requests fail there is no indication in any service or request logs of the failure. When debugging issues like connect exceptions, it can appear that the request was not retried the correct number of times or that the request was retried on the same host.

The majority of debugging is done after the fact, so debug logs are generally not very useful.

When we receive QoS responses, there is at least a request log in the remote service. So those logs probably should stay debug.